contrib/agent/src/integration-test/java/io/opencensus/contrib/agent/instrumentation/ThreadInstrumentationIT.java

```diff
@@ -115,33 +113,35 @@ public void execute(Runnable command) {
             // been wrapped in a different context (by automatic instrumentation of
             // Executor#execute), that context will be attached when executing the Runnable.
             Context context2 = Context.current().withValue(KEY, "wrong context");
-            // Work around findbugs warning. Context.attach() is marked as @CheckReturnValue so we
-            // need to check the return
-            // value here, otherwise findbugs will fail.
-            Preconditions.checkNotNull(context2.attach(), "context2.attach()");
-            Thread thread = new Thread(command);
-            thread.start();
+            Context context3 = context2.attach();
             try {
-              thread.join();
-            } catch (InterruptedException ex) {
-              Thread.currentThread().interrupt();
+              Thread thread = new Thread(command);
+              thread.start();
+              try {
+                thread.join();
+              } catch (InterruptedException ex) {
+                Thread.currentThread().interrupt();
+              }
+            } finally {
+              context2.detach(context3);
             }
-            context2.detach(context);
           }
         };
 
+    final AtomicReference<Context> newThreadCtx = new AtomicReference<Context>();
+    final AtomicReference<String> newThreadCtxVal = new AtomicReference<String>();
     newThreadExecutor.execute(
         new Runnable() {
           @Override
           public void run() {
-            // Assert that the automatic context propagation added by ThreadInstrumentation did not
-            // interfere with the automatically propagated context from Executor#execute.
-            assertThat(Context.current()).isSameAs(context);
-            assertThat(KEY.get()).isEqualTo("myvalue");
-            tested.set(true);
+            newThreadCtx.set(Context.current());
+            newThreadCtxVal.set(KEY.get());
           }
         });
 
-    assertThat(tested.get()).isTrue();
+    // Assert that the automatic context propagation added by ThreadInstrumentation did not
+    // interfere with the automatically propagated context from Executor#execute.
+    assertThat(newThreadCtx.get()).isSameAs(context);
+    assertThat(newThreadCtxVal.get()).isEqualTo("myvalue");
```

Comment:
This is not necessary, because Context is immutable.

Reply:
I see, removed. (It's a copy-and-paste from original test.)